cypress/e2e/po/pages/extensions.po.ts

```diff
@@ -0,0 +1,249 @@
+  extensionTabs() {
+    return this.self().getId('extension-tabs');
+  }
+
+  extensionTabInstalledClick(): Cypress.Chainable {
+    return this.extensionTabs().get('li:nth-child(1) a').click();
+  }
+
+  extensionTabAvailableClick(): Cypress.Chainable {
+    return this.extensionTabs().get('li:nth-child(2) a').click();
+  }
+
+  extensionTabUpdatesClick(): Cypress.Chainable {
+    return this.extensionTabs().get('li:nth-child(3) a').click();
+  }
+
+  extensionTabAllClick(): Cypress.Chainable {
+    return this.extensionTabs().get('li:nth-child(4) a').click();
+  }
+
+  // extension reload banner
+  extensionReloadBanner() {
+    return this.self().getId('extension-reload-banner');
+  }
+
+  extensionReloadClick(): Cypress.Chainable {
+    return this.extensionReloadBanner().getId('extension-reload-banner-reload-btn').click();
+  }
+
+  // extension menu
+  private extensionMenu() {
+    return this.self().getId('extensions-page-menu');
+  }
+
+  extensionMenuToggle(): Cypress.Chainable {
+    return this.extensionMenu().click();
+  }
+
+  manageReposClick(): Cypress.Chainable {
+    return this.self().getId('action-menu-0-item').click();
+  }
+
+  // this will only appear with developer mode enabled
+  // developerLoadClick(): Cypress.Chainable {
+  //   return this.self().getId('action-menu-2-item').click();
+  // }
+
+  disableExtensionsClick(): Cypress.Chainable {
+    return this.self().getId('action-menu-2-item').click();
+  }
+
+  // disable extensions OVERALL modal
+  disableExtensionModal() {
+    return this.self().getId('disable-ext-modal');
+  }
+
+  removeRancherExtRepoCheckboxClick(): Cypress.Chainable {
+    return this.self().getId('disable-ext-modal-remove-repo').click();
+  }
+
+  disableExtensionModalCancelClick(): Cypress.Chainable {
+    return this.disableExtensionModal().get('.dialog-buttons button:first-child').click();
+  }
+
+  disableExtensionModalDisableClick(): Cypress.Chainable {
+    return this.disableExtensionModal().get('.dialog-buttons button:last-child').click();
+  }
+
+  // enable extensions OVERALL modal
+  enableExtensionModal() {
+    return this.self().get('[data-modal="confirm-uiplugins-setup"]');
+  }
+
+  enableExtensionModalCancelClick(): Cypress.Chainable {
+    return this.enableExtensionModal().get('.dialog-buttons button:first-child').click();
+  }
+
+  enableExtensionModalEnableClick(): Cypress.Chainable {
+    return this.enableExtensionModal().get('.dialog-buttons button:last-child').click();
+  }
+
+  // install operator
+  installOperatorBtn() {
+    return this.self().getId('extension-enable-operator');
+  }
+
+  installOperatorBtnClick(): Cypress.Chainable {
+    return this.installOperatorBtn().click();
+  }
+
+  // add a new repo (Extension Examples)
+  createNewClusterRepoClick(): Cypress.Chainable {
+    return this.self().getId('masthead-create').click();
+  }
+
+  enterClusterRepoName(name: string) {
+    return cy.get('[data-testid="name-ns-description-name"] input').type(name);
```

Comment:
Name NS Description stuff can use the `cypress/e2e/po/components/name-ns-description.po.ts` PO

Reply:
done

Reply:
done